config/jupyter_server_config.py

```diff
@@ -1,16 +1,23 @@
 # Configuration file for jupyter server.
+import hashlib
+import random
+
 from os import getenv, getuid
 from pwd import getpwall
-from IPython.lib import passwd
 from traitlets.config import Config
 from djlab import get_djlab_config
 
 user = [u for u in getpwall() if u.pw_uid == getuid()][0]
 c = Config() if 'c' not in locals() else c
 
 # The password to use i.e. "datajoint".
-c.ServerApp.password = passwd(get_djlab_config(
-    'djlab.jupyter_server.password')).encode('utf-8')
+salt_len = 12
+h = hashlib.new('sha1')
+salt = ('%0' + str(salt_len) + 'x') % random.getrandbits(4 * salt_len)
+h.update(get_djlab_config(
+    'djlab.jupyter_server.password').encode('utf-8') + salt.encode('ascii'))
+
+c.ServerApp.password = ':'.join(('sha1', salt, h.hexdigest()))
```

Comment:
To clean this up a bit more in case we'd like to test this later, you could do something like the following. Also, we could *blacken* the code here too for styling.

```suggestion
def passwd(passphrase: str):
    salt_len = 12
    h = hashlib.new('sha1')
    salt = ('%0' + str(salt_len) + 'x') % random.getrandbits(4 * salt_len)
    h.update(passphrase.encode('utf-8') + salt.encode('ascii'))
    return ':'.join(('sha1', salt, h.hexdigest()))

c.ServerApp.password = passwd(get_djlab_config('djlab.jupyter_server.password'))
```